torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1263,6 +1264,59 @@ def no_sync(self) -> Generator:
                     "`True` while in the `no_sync()` context manager"
                 m._require_backward_grad_sync = old_flag
 
+    @torch.no_grad()
+    def _clip_grad_norm_(
+        self, max_norm: Union[float, int], norm_type: Union[float, int] = 2.0
+    ) -> None:
+        """
+        Clip all gradients at this point in time. The norm is computed over all
+        gradients together, as if they were concatenated into a single vector.
+        Gradients are modified in-place.
+
+        Args:
+            max_norm (float or int): max norm of the gradients
+            norm_type (float or int): type of the used p-norm. Can be ``'inf'``
+                for infinity norm.
+
+        Returns:
+            Total norm of the parameters (viewed as a single vector).
+
+        .. note:: This is analogous to `torch.nn.utils.clip_grad_norm_` but
+            handles the partitioning and multiple devices per rank under the
+            hood. The default torch util is not applicable here, because each
+            rank only has a partial view of all the grads in the model, so
+            calling it in the OSS context would lead to different scaling being
+            applied per subset of model parameters.
+
+        .. warning:: This needs to be called on all ranks, since synchronization
+            primitives will be used.
+        """
+        self._lazy_init()
```

Comment:
nit: add a comment to explain why _lazy_init() is needed

Reply:
sure.